Upstream: merged, PR 4043 in dotnet/dotnet (backported)
Reason: Fix versions calculcations when building in 2026+

From c6af058cf9396f55636f491949e5a199ca700bb0 Mon Sep 17 00:00:00 2001
From: Viktor Hofer <viktor.hofer@microsoft.com>
Date: Mon, 5 Jan 2026 11:10:29 +0100
Subject: [PATCH 1/4] Add patch for failing identitymodel submodule

---
 ...assembly-version-calculation-in-2026.patch | 25 +++++++++++++++++++
 1 file changed, 25 insertions(+)
 create mode 100644 src/source-build-externals/patches/azure-activedirectory-identitymodel-extensions-for-dotnet/0002-Fix-assembly-version-calculation-in-2026.patch

diff --git a/src/source-build-externals/patches/azure-activedirectory-identitymodel-extensions-for-dotnet/0002-Fix-assembly-version-calculation-in-2026.patch b/src/source-build-externals/patches/azure-activedirectory-identitymodel-extensions-for-dotnet/0002-Fix-assembly-version-calculation-in-2026.patch
new file mode 100644
index 00000000000..6502f25ebde
--- /dev/null
+++ b/src/source-build-externals/patches/azure-activedirectory-identitymodel-extensions-for-dotnet/0002-Fix-assembly-version-calculation-in-2026.patch
@@ -0,0 +1,25 @@
+From d9e04942a63fa650f9ca328fa5e3b3e016cfce81 Mon Sep 17 00:00:00 2001
+From: Viktor Hofer <viktor.hofer@microsoft.com>
+Date: Mon, 5 Jan 2026 11:09:12 +0100
+Subject: [PATCH] Fix assembly version calculation in 2026
+
+---
+ .../build/common.props                                          | 2 +-
+ 1 file changed, 1 insertion(+), 1 deletion(-)
+
+diff --git a/src/source-build-externals/src/azure-activedirectory-identitymodel-extensions-for-dotnet/build/common.props b/src/source-build-externals/src/azure-activedirectory-identitymodel-extensions-for-dotnet/build/common.props
+index efd7e90e2f4..11ba2a6e49f 100644
+--- a/src/source-build-externals/src/azure-activedirectory-identitymodel-extensions-for-dotnet/build/common.props
++++ b/src/source-build-externals/src/azure-activedirectory-identitymodel-extensions-for-dotnet/build/common.props
+@@ -42,7 +42,7 @@
+     <VersionSuffix Condition="'$(WilsonVersion)' == ''">$(PreviewVersionSuffix)</VersionSuffix>
+     <VersionPrefix Condition="'$(WilsonVersion)' == ''">$(WilsonCurrentVersion)</VersionPrefix>
+     <FileVersion Condition="'$(WilsonVersion)' != '' and '$(IsCustomPreview)' != 'true' ">$(WilsonVersion).$([System.DateTime]::Now.AddYears(-2019).Year)$([System.DateTime]::Now.ToString("MMdd"))</FileVersion>
+-    <FileVersion Condition="'$(WilsonVersion)' == ''">$(WilsonCurrentVersion).$([System.DateTime]::Now.AddYears(-2019).Year)$([System.DateTime]::Now.ToString("MMdd"))</FileVersion>
++    <FileVersion Condition="'$(WilsonVersion)' == ''">$(WilsonCurrentVersion).$([System.DateTime]::Now.AddYears(-2019).Year)$([System.DateTime]::Now.ToString("dd"))</FileVersion>
+   </PropertyGroup>
+ 
+   <PropertyGroup>
+-- 
+2.45.2
+

From ecebdf47999d2c49ff5db9cb57d3ec80412e6b1b Mon Sep 17 00:00:00 2001
From: Viktor Hofer <viktor.hofer@microsoft.com>
Date: Mon, 5 Jan 2026 11:44:29 +0100
Subject: [PATCH 3/4] Fix FileVersion calculation in common.props

---
 ...assembly-version-calculation-in-2026.patch | 41 +++++++++++++------
 1 file changed, 29 insertions(+), 12 deletions(-)

diff --git a/src/source-build-externals/patches/azure-activedirectory-identitymodel-extensions-for-dotnet/0002-Fix-assembly-version-calculation-in-2026.patch b/src/source-build-externals/patches/azure-activedirectory-identitymodel-extensions-for-dotnet/0002-Fix-assembly-version-calculation-in-2026.patch
index 6502f25ebde..4530538a2bf 100644
--- a/src/source-build-externals/patches/azure-activedirectory-identitymodel-extensions-for-dotnet/0002-Fix-assembly-version-calculation-in-2026.patch
+++ b/src/source-build-externals/patches/azure-activedirectory-identitymodel-extensions-for-dotnet/0002-Fix-assembly-version-calculation-in-2026.patch
@@ -1,25 +1,42 @@
-From d9e04942a63fa650f9ca328fa5e3b3e016cfce81 Mon Sep 17 00:00:00 2001
+From 22c9a70c8e0fe6a30190430d93f12fa29a304288 Mon Sep 17 00:00:00 2001
 From: Viktor Hofer <viktor.hofer@microsoft.com>
-Date: Mon, 5 Jan 2026 11:09:12 +0100
-Subject: [PATCH] Fix assembly version calculation in 2026
+Date: Mon, 5 Jan 2026 11:43:26 +0100
+Subject: [PATCH] Fix FileVersion calculation in 2026
 
 ---
- .../build/common.props                                          | 2 +-
- 1 file changed, 1 insertion(+), 1 deletion(-)
+ build/common.props | 19 ++++++++++++++++++-
+ 1 file changed, 18 insertions(+), 1 deletion(-)
 
-diff --git a/src/source-build-externals/src/azure-activedirectory-identitymodel-extensions-for-dotnet/build/common.props b/src/source-build-externals/src/azure-activedirectory-identitymodel-extensions-for-dotnet/build/common.props
-index efd7e90e2f4..11ba2a6e49f 100644
---- a/src/source-build-externals/src/azure-activedirectory-identitymodel-extensions-for-dotnet/build/common.props
-+++ b/src/source-build-externals/src/azure-activedirectory-identitymodel-extensions-for-dotnet/build/common.props
-@@ -42,7 +42,7 @@
+diff --git a/build/common.props b/build/common.props
+index efd7e90e..e5f6f3ed 100644
+--- a/build/common.props
++++ b/build/common.props
+@@ -42,7 +42,24 @@
      <VersionSuffix Condition="'$(WilsonVersion)' == ''">$(PreviewVersionSuffix)</VersionSuffix>
      <VersionPrefix Condition="'$(WilsonVersion)' == ''">$(WilsonCurrentVersion)</VersionPrefix>
      <FileVersion Condition="'$(WilsonVersion)' != '' and '$(IsCustomPreview)' != 'true' ">$(WilsonVersion).$([System.DateTime]::Now.AddYears(-2019).Year)$([System.DateTime]::Now.ToString("MMdd"))</FileVersion>
 -    <FileVersion Condition="'$(WilsonVersion)' == ''">$(WilsonCurrentVersion).$([System.DateTime]::Now.AddYears(-2019).Year)$([System.DateTime]::Now.ToString("MMdd"))</FileVersion>
-+    <FileVersion Condition="'$(WilsonVersion)' == ''">$(WilsonCurrentVersion).$([System.DateTime]::Now.AddYears(-2019).Year)$([System.DateTime]::Now.ToString("dd"))</FileVersion>
++    <!-- TODO: Makes sure that the revision is higher than what's in already shipped packages (> 61231).
++         This will overflow in 2030 though and needs a long term fix. -->
++    <FileVersion Condition="'$(WilsonVersion)' == ''">$(WilsonCurrentVersion).$([MSBuild]::Add(
++      61232,
++      $([MSBuild]::Add(
++        $([MSBuild]::Add(
++          $([MSBuild]::Multiply(
++            $([System.DateTime]::Now.AddYears(-2019).Year),
++            385
++          )),
++          $([MSBuild]::Multiply(
++            $([System.DateTime]::Now.ToString("MM")),
++            32
++          ))
++        )),
++        $([System.DateTime]::Now.ToString("dd"))
++      ))
++    ))</FileVersion>
    </PropertyGroup>
  
    <PropertyGroup>
 -- 
-2.45.2
+2.52.0.windows.1
 

From aaba42c4aaddf222388ce6d5930f9448a0136fa6 Mon Sep 17 00:00:00 2001
From: Viktor Hofer <viktor.hofer@microsoft.com>
Date: Mon, 5 Jan 2026 11:47:14 +0100
Subject: [PATCH 4/4] fix comment

---
 .../0002-Fix-assembly-version-calculation-in-2026.patch       | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/src/source-build-externals/patches/azure-activedirectory-identitymodel-extensions-for-dotnet/0002-Fix-assembly-version-calculation-in-2026.patch b/src/source-build-externals/patches/azure-activedirectory-identitymodel-extensions-for-dotnet/0002-Fix-assembly-version-calculation-in-2026.patch
index 4530538a2bf..0018b8b51ce 100644
--- a/src/source-build-externals/patches/azure-activedirectory-identitymodel-extensions-for-dotnet/0002-Fix-assembly-version-calculation-in-2026.patch
+++ b/src/source-build-externals/patches/azure-activedirectory-identitymodel-extensions-for-dotnet/0002-Fix-assembly-version-calculation-in-2026.patch
@@ -17,14 +17,14 @@ index efd7e90e..e5f6f3ed 100644
      <FileVersion Condition="'$(WilsonVersion)' != '' and '$(IsCustomPreview)' != 'true' ">$(WilsonVersion).$([System.DateTime]::Now.AddYears(-2019).Year)$([System.DateTime]::Now.ToString("MMdd"))</FileVersion>
 -    <FileVersion Condition="'$(WilsonVersion)' == ''">$(WilsonCurrentVersion).$([System.DateTime]::Now.AddYears(-2019).Year)$([System.DateTime]::Now.ToString("MMdd"))</FileVersion>
 +    <!-- TODO: Makes sure that the revision is higher than what's in already shipped packages (> 61231).
-+         This will overflow in 2030 though and needs a long term fix. -->
++         This will overflow in 2029 though and needs a long term fix. -->
 +    <FileVersion Condition="'$(WilsonVersion)' == ''">$(WilsonCurrentVersion).$([MSBuild]::Add(
 +      61232,
 +      $([MSBuild]::Add(
 +        $([MSBuild]::Add(
 +          $([MSBuild]::Multiply(
 +            $([System.DateTime]::Now.AddYears(-2019).Year),
-+            385
++            416
 +          )),
 +          $([MSBuild]::Multiply(
 +            $([System.DateTime]::Now.ToString("MM")),
